lib/puppet/file_serving/s3_metadata.rb

```diff
@@ -0,0 +1,22 @@
+require 'puppet/file_serving/metadata'
+
+class Puppet::FileServing::S3Metadata < Puppet::FileServing::Metadata
+  def initialize(s3_response, path = '/dev/null')
+    super(path)
+
+    # ignore options that do not apply to s3 metadata
+    @owner = @group = @mode = nil
+
+    if s3_response.etag
+      @checksum_type = 'md5'
+      # Strip the double quotes from the S3 etag response
+      @checksum = "{md5}#{s3_response.etag.tr('"','')}"
```

Comment:
So does the `Etag` header as used by S3 consist of the actual MD5 hash of the file's content? Otherwise this won't work.

Perhaps @garethr or @Iristyle can elaborate?